packages/opentelemetry-tracing/test/export/TestTracingSpanExporter.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  BasicTracerProvider,
+  SpanExporter,
+  ReadableSpan,
+  Tracer,
+  SpanProcessor,
+} from '../../src';
+import { ExportResult, NoopLogger, AlwaysOnSampler } from '@opentelemetry/core';
+
+/**
+ * A test-only span exporter that naively simulates triggering instrumentation
+ * (creating new spans) during export.
+ */
+export class TestTracingSpanExporter implements SpanExporter {
+  private _processedSpans: ReadableSpan[] = [];
+  private _exporterCreatedSpans: ReadableSpan[] = [];
+  private _stopped = false;
+  private _tracer: Tracer;
+
+  constructor() {
+    const tracerProvider = new BasicTracerProvider({
+      logger: new NoopLogger(),
+    });
+
+    const spanProcessor: SpanProcessor = {
+      forceFlush: () => {},
+      onStart: () => {},
+      shutdown: () => {},
+      onEnd: span => {
+        this._exporterCreatedSpans.push(span);
+      },
+    };
+
+    tracerProvider.addSpanProcessor(spanProcessor);
+
+    this._tracer = new Tracer(
+      { name: 'default', version: '0.0.1' },
+      { sampler: new AlwaysOnSampler() },
+      tracerProvider
+    );
+  }
+
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ): void {
+    if (this._stopped) {
+      return resultCallback(ExportResult.FAILED_NOT_RETRYABLE);
+    }
+
+    // Simulates an instrumented exporter by creating a span on the tracer.
+    const createdSpan = this._tracer.startSpan('exporter-created-span');
+    createdSpan.end();
+
+    this._processedSpans.push(...spans);
+    return resultCallback(ExportResult.SUCCESS);
```

Comment:
The real export methods will call the callback asynchronously. This could result in a situation where tests pass but the real system behaves differently.

```suggestion
    setTimeout(() => resultCallback(ExportResult.SUCCESS), 0);
```

Reply:
oh, good call-out. i somewhat blindly followed the in-memory exporter.

Reply:
The in-memory exporter should probably also be fixed :)